File: bms/go/ach/moov_io_ach_ADVBatchControl_EntryAddendaCountField/moov_io_ach_ADVBatchControl_EntryAddendaCountField.go
package test

import (
	"strconv"
	"strings"
)

var (
	moov_io_ach_stringZeros map[int]string = moov_io_ach_populateMap(94, "0")
)

const ()

func (bc *moov_io_ach_ADVBatchControl) EntryAddendaCountField() string {
	return bc.numericField(bc.EntryAddendaCount, 6)
}

func (c *moov_io_ach_converters) numericField(n int, max uint) string {
	s := strconv.FormatInt(int64(n), 10)
	if l := uint(len(s)); l > max {
		return s[l-max:]
	} else {
		m := int(max - l)
		pad, exists := moov_io_ach_stringZeros[m]
		if exists {
			return pad + s
		}
		// slow path
		return strings.Repeat("0", m) + s
	}
}

func moov_io_ach_populateMap(max int, zero string) map[int]string {
	out := make(map[int]string, max)
	for i := 0; i < max; i++ {
		out[i] = strings.Repeat(zero, i)
	}
	return out
}

type moov_io_ach_ADVBatchControl struct {
	// ID is a client defined string used as a reference to this record.

	// This should be the same as BatchHeader ServiceClassCode for ADV: AutomatedAccountingAdvices.

	// EntryAddendaCount is a tally of each Entry Detail Record and each Addenda
	// Record processed, within either the batch or file as appropriate.
	EntryAddendaCount int `json:"entryAddendaCount"`
	// validate the Receiving DFI Identification in each Entry Detail Record is hashed
	// to provide a check against inadvertent alteration of data contents due
	// to hardware failure or program error
	//
	// In this context the Entry Hash is the sum of the corresponding fields in the
	// Entry Detail Records on the file.

	// TotalDebitEntryDollarAmount Contains accumulated Entry debit totals within the batch.

	// TotalCreditEntryDollarAmount Contains accumulated Entry credit totals within the batch.

	// ACHOperatorData is an alphanumeric code used to identify an ACH Operator

	// ODFIIdentification the routing number is used to identify the DFI originating entries within a given branch.

	// BatchNumber this number is assigned in ascending sequence to each batch by the ODFI
	// or its Sending Point in a given file of entries. Since the batch number
	// in the Batch Header Record and the Batch Control Record is the same,
	// the ascending sequence number should be assigned by batch and not by record.

	// validator is composed for data validation

	// converters is composed for ACH to golang Converters
	moov_io_ach_converters
}

type moov_io_ach_converters struct{}

File: translations_only/success/go/ach/moov_io_ach_ADVBatchControl_EntryAddendaCountField/gpt4/Restart/moov_io_ach_ADVBatchControl_EntryAddendaCountField.rs

use std::collections::HashMap;

lazy_static! {
    static ref MOOV_IO_ACH_STRING_ZEROS: HashMap<usize, String> = moov_io_ach_populate_map(94, "0");
}

fn moov_io_ach_populate_map(max: usize, zero: &str) -> HashMap<usize, String> {
    let mut out = HashMap::with_capacity(max);
    for i in 0..max {
        out.insert(i, zero.repeat(i));
    }
    out
}

struct MoovIoAchADVBatchControl {
    entry_addenda_count: i32,
}

impl MoovIoAchADVBatchControl {
    fn entry_addenda_count_field(&self) -> String {
        MoovIoAchConverters::numeric_field(self.entry_addenda_count, 6)
    }
}

struct MoovIoAchConverters;

impl MoovIoAchConverters {
    fn numeric_field(n: i32, max: usize) -> String {
        let s = n.to_string();
        let l = s.len();
        if l > max {
            s[l - max..].to_string()
        } else {
            let m = max - l;
            if let Some(pad) = MOOV_IO_ACH_STRING_ZEROS.get(&m) {
                format!("{}{}", pad, s)
            } else {
                // slow path
                "0".repeat(m) + &s
            }
        }
    }
}

impl MoovIoAchADVBatchControl {
    fn new(entry_addenda_count: i32) -> Box<Self> {
        Box::new(Self {
            entry_addenda_count,
            ..Default::default()
        })
    }
}

impl Default for MoovIoAchADVBatchControl {
    fn default() -> Self {
        Self {
            entry_addenda_count: 0,
        }
    }
}

impl Default for MoovIoAchConverters {
    fn default() -> Self {
        Self {}
    }
}

#[macro_use]
extern crate lazy_static;
